src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -3340,15 +3399,26 @@ export class ResultsViewPageStore extends AnalysisStore
             // filters can be an empty list
             // when all selected samples are coming from studies that don't have structural variant profile
             // in this case, we should not fetch structural variants data
-            if (_.isEmpty(filters)) {
+            if (_.isEmpty(sampleMolecularIdentifiers)) {
                 return [];
             } else {
+                const entrezGeneIds = genes
```

Comment:
I think genes is, like 20k long, so perhaps we should have a computed lookup which is symbol to gene and then you can just iterate through structVarHugoGeneSymbols

Reply:
I think that `genes` is the genes that are derived from the query. There is a structure called `referenceGenes` that appears to represent all genes in the reference genome (~20k). Howerer, I have I agree that this mapping would be more efficient with a looup. I have created a hugoGeneSymbol => Gene lookup and mapped from there.